class-wc-gateway-komoju.php

```diff
@@ -14,186 +14,224 @@
+        $data = $this->get_input_field_data();
+        $method_fields = array( 'method-field' => $data );
+        $fields = wp_parse_args( $fields, apply_filters( 'woocommerce_komoju_method_form_fields', $method_fields, $this->id ) );
         ?>
         <fieldset id="<?php echo $this->id; ?>-cc-form">
-            <?php do_action( 'woocommerce_komoju_method_form_start', $this->id ); ?>
-            <?php
-                foreach ( $fields as $field ) {
-                    echo $field;
-                }
-            ?>
-            <?php do_action( 'woocommerce_komoju_method_form_end', $this->id ); ?>
-            <div class="clear"></div>
+          <?php do_action( 'woocommerce_komoju_method_form_start', $this->id ); ?>
+          <?php
+            foreach ( $fields as $field ) {
+                echo $field;
+            }
+          ?>
+          <?php do_action( 'woocommerce_komoju_method_form_end', $this->id ); ?>
+          <div class="clear"></div>
         </fieldset>
         <?php
-    }	
+    }
 
-	private function get_mydefault_api_url(){
-		// In dev the relative plugin URL will remove the host name, but it
-		// will appear in production instances
-		return WC()->api_request_url( 'WC_Gateway_Komoju' );
-	}
+    private function get_mydefault_api_url(){
+        // In dev the relative plugin URL will remove the host name, but it
+        // will appear in production instances
+        return WC()->api_request_url( 'WC_Gateway_Komoju' );
+    }
 
-	/**
+    private function get_input_field_data() {
+        $komoju_client = new KomojuApi( $this->secretKey );
+
+        try {
+            $methods = $komoju_client->paymentMethods();
+            $page_locale = $this->get_locale_or_fallback();
+            $name_property =  "name_{$page_locale}";
+
+            $field_data = '
+                <p
+                  class="
+                    form-row
+                    form-row-wide
+                    validate-required
+                    woocommerce-validated"
+                >
+                <label
+                  for="' . esc_attr( $this->id ) . '-method"
+                >' . __( 'Method of payment:', 'komoju-woocommerce' ) . '
+                  <abbr
+                    class="required"
+                    title="required"
+                  >*
+                  </abbr>
+               </label>';
+            foreach ($methods as $method) {
+              $field_data.= '
+                  <input
+                    id="' . esc_attr( $this->id) . '-method"
+                    class="input-radio"
+                    type="radio"
+                    value="'. esc_attr( $method->type_slug ) .'"
+                    name="' . esc_attr( $this->id). '-method"
+                  />
+                  '. ( $method->{$name_property} ) .'
+                  <br/>';
+            }
+            $field_data .= '</p>';
+        } catch (KomojuExceptionBadServer $e) {
```

Comment:
Looking at the php library it can also throw a `KomojuExceptionBadJson`, do you think we should also be catching and handling that here?

Reply:
Thanks for catching that - I added the other exception class.